### PR TITLE
Small: Fix NumPy 2.0 test script breakage

### PR DESCRIPTION
**Pull request by BenWibking**
## PR Summary

This fixes the test script breakage due to NumPy 2.0 API changes (https://numpy.org/devdocs/numpy_2_0_migration_guide.html#numpy-2-0-migration-guide)

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py [hunk 1]
```diff
@@ -648,11 +648,11 @@ def Get(self, variable, flatten=True, interior=False, average_to_cell_centers=Tr
         if flatten:
             nblocks = vShape[0]
             if self.varTopology[variable] == "None":
-                remaining_size = np.product(vShape[1:])
+                remaining_size = np.prod(vShape[1:])
```
Comment (lroberts36) [inline scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py:651]: glad to see that the numpy gods thought this was a necessary change.

**Review by lroberts36 — APPROVED** [review]
LGTM, thanks for figuring this out.

File: scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py (hunk 1 above)
Comment (lroberts36) [reply scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py:651]: actually, does `prod` work for older versions of numpy or will this break the test suite when not using the newer numpy?

Comment (lroberts36) [reply scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py:651]: Looks like the change should be backwards compatible: https://stackoverflow.com/questions/49863633/numpy-product-vs-numpy-prod-vs-ndarray-prod.

**Review by pgrete — APPROVED** [review]

**Pull request merged**